commands/container-registry/registry/registry_test.go

```diff
@@ -0,0 +1,84 @@
+package registry
+
+import (
+	"context"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/config"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	ionoscloud "github.com/ionos-cloud/sdk-go-container-registry"
+	"github.com/lucasjones/reggen"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestRegistryService(t *testing.T) {
+	t.Run("registry functions", func(t *testing.T) {
+		viper.Reset()
+		viper.Set(constants.ArgOutput, constants.DefaultOutputFormat)
+		viper.Set(constants.ArgQuiet, false)
+		viper.Set(constants.ArgVerbose, false)
+		viper.Set(constants.ArgForce, true)
+
+		name, err := reggen.Generate("^[a-z][-a-z0-9]{1,61}[a-z0-9]$", 10)
+		assert.NoError(t, err)
+		c := RegPostCmd()
+		c.Command.Flags().Set(FlagName, name)
+		c.Command.Flags().Set(FlagLocation, "de/fra")
+
+		err = c.Command.Execute()
+		assert.NoError(t, err)
+
+		svc, err := config.GetClient()
+		registries, _, err := svc.RegistryClient.RegistriesApi.RegistriesGet(context.Background()).Execute()
+		assert.NoError(t, err)
+
+		var newRegistry *ionoscloud.RegistryResponse
+		for _, registry := range *registries.GetItems() {
+			if *registry.GetProperties().GetName() == name {
+				newRegistry = &registry
+			}
+		}
+
```

Comment:
I think it would be great if we could add an assert that either: newRegistry is not nil (i.e. the if-block was entered least once), or that newRegistry's name is `name` - this second option would cause a panic though, if the if-block isn't entered at at all

Reply:
sorry. my bad, we're using `newRegistry` below. this works fine. Nice!